notebooks/2.single-cell-compound-prio/4.selecting_hits.py

```diff
@@ -0,0 +1,218 @@
+    else:
+        raise ValueError(f"Method {method} not recognized")
+
+    # next we save the scores into a list
+    aggregated_score = [
+        strata_name[0],
+        off_morph_score,
+        on_morph_score,
+        comb_morph_score,
+    ]
+    scores.append(aggregated_score)
+
+# convert scores to a dataframe
+scores = pd.DataFrame(
+    scores,
+    columns=["treatment_name", "off_morph_score", "on_morph_score", "combined_score"],
+)
+ranked_df = scores.sort_values(
+    by=["on_morph_score", "off_morph_score"], ascending=[False, False]
+)
+ranked_df = ranked_df.reset_index(drop=True)
+
+# next we use the .rank() method to rank the scores where the lowest value is the highest rank
+ranked_df["on_morph_rank"] = ranked_df["on_morph_score"].rank(ascending=False)
+ranked_df["off_morph_rank"] = ranked_df["off_morph_score"].rank(ascending=False)
+ranked_df["combined_morph_rank"] = ranked_df[["on_morph_rank", "off_morph_rank"]].mean(
+    axis=1
+)
+
+# display
+print(ranked_df.shape)
+ranked_df.head()
+
+
+# In[6]:
+
+
+# Next we want to merge the pathway information with the ranked_df
+# First we need to update the DMSO name based on the cell type to match with the DMSO
+# label in the ranked_df
+pathway_df["UCD ID"] = pathway_df.apply(
+    lambda row: update_dmso_name(row["UCD ID"], row["Cell Type"]), axis=1
+)
+
+# Select only the UCD ID and Pathway columns
+pathway_info_df = pathway_df[["UCD ID", "Pathway"]]
+
+# merge the pathway info with the ranked_df
+ranked_df = ranked_df.merge(
+    pathway_info_df, left_on="treatment_name", right_on="UCD ID", how="left"
+)
+
+# Drop the UCD ID column
+ranked_df = ranked_df.drop(columns=["UCD ID"])
+
+# display
+print(ranked_df.shape)
+ranked_df.head()
+
+
+# Next we rank they treatments based on their three scores
+
+# In[7]:
+
+
+# Melt the DataFrame for plotting
+# This transforms the DataFrame from wide format to long format
+# so that we can plot the ranks for different treatments and pathways
+ranked_df_melted = ranked_df.melt(
+    id_vars=["treatment_name", "Pathway"],
+    value_vars=["on_morph_rank", "off_morph_rank", "combined_morph_rank"],
+    var_name="Rank Type",
+    value_name="Rank",
+)
+
+# Plot using seaborn
+# Here we are plotting the ranks for different treatments and pathways
+# The x-axis is the treatment name, the y-axis is the rank, the hue is the pathway,
+# and the style is the rank type
+plt.figure(figsize=(12, 11))
+sns.scatterplot(
+    data=ranked_df_melted,
+    x="treatment_name",
+    y="Rank",
+    hue="Pathway",
+    style="Rank Type",
+    s=100,
+)
+
+# Updating the axis labels and title
+plt.xticks(rotation=90, fontsize=15)
+plt.yticks(fontsize=15)
+
+# Adding a title, labels and legend
+plt.title("Ranks for Different Treatments", fontsize=30)
+plt.xlabel("Treatment Name", fontsize=20)
+plt.ylabel("Rank", fontsize=20)
+
+# adding a legend
+plt.legend()
+
+# Saving the plot
+plt.tight_layout()
+plt.show()
```

Comment:
Recommend saving this plot, it is really cool!

Reply:
Done!